### PR TITLE
Feature/roc 9582

### PR DESCRIPTION
**Pull request by jaberelfasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 1]
```diff
@@ -95,6 +96,13 @@ private CallbackResponse handleDefendAllClaim(CallbackParams callbackParams) {
                 .build();
         }
         if ("DEFENDANT_RESPONSE_SPEC".equals(callbackParams.getRequest().getEventId())) {
+
+            if (caseData.getRespondToClaim().getHowMuchWasPaid().compareTo(caseData.getTotalClaimAmount())<0){
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:100]: A null check should be done on caseData.getRespondToClaim().getHowMuchWasPaid()

Also could we have a space on either side of the conditional operator please, just makes reading easier

[the remark this answers is not in the document]
Comment (jaberelfasi) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java]: Done

Comment (jaberelfasi) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java]: Done

**Review by sampankumar — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 2]
```diff
@@ -95,6 +96,13 @@ private CallbackResponse handleDefendAllClaim(CallbackParams callbackParams) {
                 .build();
         }
         if ("DEFENDANT_RESPONSE_SPEC".equals(callbackParams.getRequest().getEventId())) {
+
+            if (caseData.getRespondToClaim().getHowMuchWasPaid() != null && caseData.getRespondToClaim().getHowMuchWasPaid().compareTo(caseData.getTotalClaimAmount()) < 0){
+                caseData = caseData.toBuilder().respondent1ClaimResponsePaymentAdmissionForSpec(RespondentResponseTypeSpec.PART_ADMISSION).build();
```
Comment (jaberelfasi) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:101]: caseData.toBuilder().respondent1ClaimResponsePaymentAdmissionForSpec(RespondentResponseTypeSpec.PART_ADMISSION).build();

define a Boolean instead to compare with

**Pull request closed without merging**